cmd/kubeadm/app/cmd/init.go

```diff
@@ -163,14 +162,19 @@ func NewCmdInit(out io.Writer) *cobra.Command {
 	options.bto.AddTokenFlag(cmd.PersistentFlags())
 	options.bto.AddTTLFlag(cmd.PersistentFlags())
 
+	initData, initDataErr := newInitData(cmd, options)
+	kubeadmutil.CheckErr(initDataErr)
+
 	// initialize the workflow runner with the list of phases
 	initRunner.AppendPhase(phases.NewPreflightMasterPhase())
+	// TODO: a couple of phases should go here.
+	initRunner.AppendPhase(phases.NewControlPlanePhase(initData.cfg))
```

Comment:
InitData are run time data; it confuses me to get those data passed when cmd are created (and I'm not 100% sure this will behave the same)